### ISSUE TITLE
[FEATURE] Support Additional Ways of Stopping a training

### ISSUE THREAD

**Issue by daden-ms** [microsoft/nlp-recipes#140]
### Description
Currently the repo only support stopping a training when the number of epochs is reached.
It would be great to also support stopping a trainng when 
1. a desired loss is reached
2. validation loss is greater than the training loss, like: https://github.com/Bjarten/early-stopping-pytorch
